client/js/render.js

```diff
@@ -54,11 +43,8 @@ function appendMessage(container, chanId, chanType, msg) {
 
 	// Insert date marker if date changed compared to previous message
 	if (prevMsgTime.toDateString() !== msgTime.toDateString()) {
-		lastChild.after(templates.date_marker({msgDate: msgTime}));
-
-		// If date changed, we don't need to do condensed logic
-		container.append(renderedMessage);
-		return;
+		lastChild = $(templates.date_marker({msgDate: msgTime}));
```

Comment:
Yeah, this could be the issue. It's certainly not a great plan just now.

Reply:
What do you mean?

Reply:
Sorry, that's a very confusing sentence.

I do this the code that was here could be the issue we were seeing with @Jay2k1 and what we were doing doesn't actually make any sense to do, because we are already checking if the last thing is a condensed message anyway. 

But I have just realised why this won't work actually.

https://github.com/thelounge/lounge/blob/98e3bd0ca278fbcd8b120fbdaf6dfe4ec58633e7/client/views/date-marker.tpl

The date marker isn't a ".msg". So we'll miss the date marker, condense into the previous condensed, and the order will be wrong, no?

Reply:
That is possible.. I will look further into it later.